src/store/transaction-details/transaction-details.thunks.js

```diff
@@ -47,7 +47,13 @@ function fetchTransaction(transactionIdOrHash) {
 
     return transactionPromise
       .then((res) => {
-        if (res.fromBJJ !== wallet.publicKeyBase64 && res.toBJJ !== wallet.publicKeyBase64) {
+        const txContainsBJJUserAddress =
+          res.fromBJJ !== wallet.publicKeyBase64 && res.toBJJ !== wallet.publicKeyBase64;
+        const txContainsHezEthereumAddressUserValue =
+          res.fromHezEthereumAddress !== wallet.hermezEthereumAddress &&
+          res.toHezEthereumAddress !== wallet.hermezEthereumAddress;
+
+        if (isHezEthereumAddressUserWallet && isHezEthereumAddressUserWallet) {
```

Comment:
I love the names of the vars! It would be great though using them in the conditional! 😅 What is `isHezEthereumAddressUserWallet`?

Something else: Isn't now the logic of each var inverted? I mean, `txContainsBJJUserAddress` means that the transaction DOES contain the user address, so it feels the condition should be `res.fromBJJ === wallet.publicKeyBase64 || res.toBJJ === wallet.publicKeyBase64`.

On the other hand, the right name for the condition `res.fromBJJ !== wallet.publicKeyBase64 && res.toBJJ !== wallet.publicKeyBase64` is more like `txDoesNotContainsBJJUserAddress`.

Does this make sense?

Since truthy conditions help with human comprehension, I prefer the option:
```typescript
const txContainsBJJUserAddress = res.fromBJJ === wallet.publicKeyBase64 || res.toBJJ === wallet.publicKeyBase64;
```
Then, assuming we only had to check this condition, the `if` could be:
```typescript
if (txContainsBJJUserAddress === false) { // we send the user to "/" }
```

But we need to check 2 conditions and regarding this I think: shouldn't we send the user to `"/"` **either** when `txContainsBJJUserAddress` or `txContainsHezEthereumAddressUserValue` is `true`?

Reply:
It's perfect now! Much easier to follow and reason about!